server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1365,7 +1365,9 @@ private Exception tryAcquireInFlightDocs(Operation operation, int addingDocs) {
         assert operation.origin() == Operation.Origin.PRIMARY : operation;
         assert operation.seqNo() == SequenceNumbers.UNASSIGNED_SEQ_NO : operation;
         assert addingDocs > 0 : addingDocs;
-        final long totalDocs = indexWriter.getPendingNumDocs() + inFlightDocCount.addAndGet(addingDocs);
+        //TODO: needs to be adapted to Lucene86?
+        //indexWriter.getPendingNumDocs() + inFlightDocCount.addAndGet(addingDocs);
+        final long totalDocs = inFlightDocCount.addAndGet(addingDocs);
```

Comment:
@dnhatn In case we need to perform this action, I am not sure what we need to here?

Reply:
I see two options for this: revert https://github.com/elastic/elasticsearch/pull/63273 in 7.10 or manually keep track of the pending documents in Elasticsearch. I prefer the first option as it's error-prone to track the pending documents in Elasticsearch. @DaveCTurner WDYT?

Reply:
I never the commit, could you please check?

Reply:
+1 to revert, deferring this bugfix is the right thing to do IMO